Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -3709,7 +3714,10 @@ def _http_request(self, method, url_suffix, full_url=None, headers=None, auth=No
                     try:
                         # Try to parse json error response
                         error_entry = res.json()
-                        err_msg += '\n{}'.format(json.dumps(error_entry))
+                        if error_handler:
+                            err_msg += error_handler(error_entry)
+                        else:
+                            err_msg += '\n{}'.format(json.dumps(error_entry))
```

Comment:
If you want to have the option to use some sort of error handler then put it right under `if not self._is_status_code_valid(res, ok_codes):`
Such that 
```
if not self._is_status_code_valid(res, ok_codes):
    if error_handler:
        return error_handler(res)
```
```